src/ansible_creator/compat.py

```diff
@@ -0,0 +1,18 @@
+"""Compat library for ansible-creator.
+
+This contains compatibility definitions for older python
+When we need to import a module differently depending on python versions, we do it
+here.
+"""
+
+# ruff: noqa: F401
```

Comment:
I think that the usual approach for these is to do the import and catch ImportError instead of using sys version.

Both ruff and pylint should be able to detect the logic and not need noqa. Even if one might need it, add it just before the line needing it and not at the top of the file as this might have different side effects.

Hint: grep for `except ImportError` in our code, you will see the pattern.

Reply:
Thinking of it, I'm not fully sure of the benefit it provides over introspecting Python versions. Even with this approach, we need both the ignores to not hit the warnings for Py 3.10 and Py >=3.11.

```code
try:
    # ruff: noqa: F401 # pylint: disable=unused-import
    from importlib.resources.abc import Traversable
except ImportError:
    # ruff: noqa: F401 # pylint: disable=unused-import
    from importlib.abc import Traversable
```

Reply:
Also, this way seems to trigger mypy (which is expected):

```bash
(py310) ➜  ansible-creator git:(custom_templates) ✗ mypy --install-types
src/ansible_creator/compat.py:12: error: Cannot find implementation or library stub for module named "importlib.resources.abc"  [import-not-found]
```

Reply:
OK.

Reply:
I see this pattern being used in ansible-navigator (https://github.com/ansible/ansible-navigator/blob/main/src/ansible_navigator/utils/compatibility.py) and widely used in ansible-core.